MapboxNavigation/NavigationMapView.swift

```diff
@@ -1185,20 +1247,50 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
      
      This method coalesces consecutive line segments that have the same congestion level.
      
-     - coordinates: The coordinates of a leg.
-     - congestions: The congestion levels along a leg. There should be one fewer congestion levels than coordinates.
+     For each item in the`CongestionSegment` collection a `CongestionLevel` substitution will take place that has a road class contained in the `trafficOverrideRoadClasses` collection.
+     For each of these items the `CongestionLevel` for `.unknown` traffic congestion will be replaced with the `.low` traffic congestion.
+     
+     - parameter coordinates: The coordinates of a leg.
+     - parameter congestions: The congestion levels along a leg. There should be one fewer congestion levels than coordinates.
+     - parameter roadClasses: A collection of road classes for each geometry index in `Intersection`. There should be the same amount of `roadClasses` and `congestions`.
+     - parameter trafficOverrideRoadClasses: A collection of road classes for which a `CongestionLevel` substitution should occur.
+     - returns: A list of `CongestionSegment` tuples with coordinate and congestion level.
      */
-    func combine(_ coordinates: [CLLocationCoordinate2D], with congestions: [CongestionLevel]) -> [CongestionSegment] {
+    func combine(_ coordinates: [CLLocationCoordinate2D],
+                 with congestions: [CongestionLevel],
+                 roadClasses: [RoadClasses?]? = nil,
+                 trafficOverrideRoadClasses: [RoadClasses]? = nil) -> [CongestionSegment] {
         var segments: [CongestionSegment] = []
         segments.reserveCapacity(congestions.count)
+        
+        var index = 0
         for (firstSegment, congestionLevel) in zip(zip(coordinates, coordinates.suffix(from: 1)), congestions) {
             let coordinates = [firstSegment.0, firstSegment.1]
-            if segments.last?.1 == congestionLevel {
+            
+            var overriddenCongestionLevel = congestionLevel
+            if let roadClasses = roadClasses,
+               let trafficOverrideRoadClasses = trafficOverrideRoadClasses,
+               roadClasses.indices.contains(index),
+               let roadClass = roadClasses[index],
+               congestionLevel == .unknown {
+                
+                for element in trafficOverrideRoadClasses {
+                    if roadClass.contains(element) {
+                        overriddenCongestionLevel = .low
+                        break
+                    }
+                }
```

Comment:
This can also be more functional, by calling `contains(where:)` on `trafficOverrideRoadClasses`.

Reply:
Will address it when `Road Classifications` from `mapbox_streets_v8` is available. There might be no need in storing `trafficOverrideRoadClasses` array.